text/0148-sampling-probability.md

```diff
@@ -0,0 +1,744 @@
+sample from those spans with attribute `B=2`.  Any span that has both
+of these properties is eligible to be selected for both samples, in
+which case one event could have two non-zero adjusted counts (e.g.,
+`sampling.by_a.adjusted_count` and `sampling.by_b.adjusted_count`).
+
+## Proposed Tracing specification
+
+For the standard OpenTelemetry Span Sampler implementations to support
+a range of probability sampling schemes, this document recommends the
+use of a Span attribute named `sampling.sampler_name.adjusted_count`
+to encode an unbiased adjusted count computed by a Sampler.
+
+The value any attribute name prefixed with `sampling.` and suffixed
+with `.adjusted_count` under this proposal MUST be an unbiased
+estimate of the total population count represented by the individual
+event.
+
+### Suggested text
+
+After
+[Sampler](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#sampler)
+and before [Builtin
+Samplers](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#built-in-samplers),
+a new section will introduce the term "adjusted count" and relate it
+to inclusion probability.  For example:
+
+```markdown
+### Probability sampling
+
+Probability Samplers are Samplers that output statistically unbiased
+inclusion probability.  Inclusion probability in the context of
+tracing is the *effective* probability of the Sampler returning
+`RECORD_AND_SAMPLE` when invoked, which can be decided locally or 
+derived from the context when the [W3C Trace Context is-sampled
+flag](https://www.w3.org/TR/trace-context/#sampled-flag) is in use.
+
+#### Adjusted count span attribute
+
+The recommended way to convey inclusion probability *for events*
+sampled in OpenTelemetry is in the form of an **adjusted count**,
+which is the reciprocal (i.e., mathematical inverse function) of
+inclusion probability.
+
+The implied goal of probability sampling is to support estimating the
+count of spans in the population using the spans that were sampled.
+The adjusted count associated with a Span is the expected value of the
+number of identical Spans within the population that each Span
+represents.  Probability samplers SHOULD ensure the samples they
+compute are unbiased, which implies that the expected value of the sum
+of adjusted counts in the sample equals the true count of spans in the
+population.
+
+Attributes used to express the adjusted count in an unbiased
+probability sampling scheme SHOULD use a Span attribute name with
+prefix `sampling.` and with suffix `.adjusted_count` (e.g.,
+`sampling.sampler_name.adjusted_count`).  Adjusted count attributes
+MAY be integer or floating-point values.
+
+#### Inclusion probability tracestate value
+
+The recommended way to convey sampling probability *for contexts* in
+OpenTelemetry is through the W3C Trace Context tracestate using the
+key `head_probability`.
```

Comment:
I think this is a common misunderstanding of tracestate header: it is not meant for arbitrary keys that have shared semantics. It is meant for vendor data.

What would be a valid use of tracestate is something like otel=headprob:0.123

Reply:
I would be reluctant to bump the version of traceparent just to capture this value, since this can be done without changing w3c spec via `tracestate: otel=headprob:0.123`.

Reply:
I see, thanks. :+1: Can we say "using the key `otel` and sub-key `headprob`" or something similar?

Reply:
> Can we say "using the key otel and sub-key headprob" or something similar?

yes